backend/dataall/modules/shares_base/services/sharing_service.py

```diff
@@ -373,103 +424,26 @@ def _get_share_data_and_items(session, share_uri, status, healthStatus=None):
         return share_data, share_items
 
     @staticmethod
-    def acquire_lock(dataset_uri, session, share_uri):
-        """
-        Attempts to acquire a lock on the dataset identified by dataset_id.
-
-        Args:
-            dataset_uri: The ID of the dataset for which the lock is being acquired.
-            session (sqlalchemy.orm.Session): The SQLAlchemy session object used for interacting with the database.
-            share_uri: The ID of the share that is attempting to acquire the lock.
-
-        Returns:
-            bool: True if the lock is successfully acquired, False otherwise.
-        """
-        try:
-            # Execute the query to get the DatasetLock object
-            dataset_lock = (
-                session.query(DatasetLock)
-                .filter(and_(DatasetLock.datasetUri == dataset_uri, ~DatasetLock.isLocked))
-                .with_for_update()
-                .first()
-            )
-
-            # Check if dataset_lock is not None before attempting to update
-            if dataset_lock:
-                # Update the attributes of the DatasetLock object
-                dataset_lock.isLocked = True
-                dataset_lock.acquiredBy = share_uri
-
-                session.commit()
-                return True
-            else:
-                log.info('DatasetLock not found for the given criteria.')
-                return False
-        except Exception as e:
-            session.expunge_all()
-            session.rollback()
-            log.error('Error occurred while acquiring lock:', e)
-            return False
-
-    @staticmethod
-    def acquire_lock_with_retry(dataset_uri, session, share_uri):
+    def acquire_lock_with_retry(
```

Comment:
this looks like a useful method, shouldn't we move it in the ResourceLockRepo?

Reply:
moved